File: 371/sum-of-two-integers.go
package main

import "fmt"

func getSum(a int, b int) int {
	if b == 0 {
		return a
	} else {
		return getSum(a^b, (a&b)<<1)
	}
}

func main() {
	a := -1
	b := 1

	var getSum func(int, int) int

	getSum = func(a int, b int) int {
		if b == 0 {
			return a
		} else {
			return getSum(a^b, (a&b)<<1)
		}
	}
	fmt.Println(getSum(a, b))
}

File: 371/sum-of-two-integers.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var getSum = function (a, b) {
    if (b == 0)
        return a;
    return getSum(a ^ b, (a & b) << 1);
};

// console.log(getSum(1, 2));
console.log(getSum(1, -1));
console.log(getSum(3 ,4));
console.log(getSum(3, 5));
console.log(getSum(10, 30));
